nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/operations/TestColdStorageRendition.java

```diff
@@ -0,0 +1,181 @@
+        Rendition rendition = renditionService.getRendition(document, "Small");
+        Blob expectedBlob = rendition.getBlob();
+
+        moveContentToColdStorage(session, document);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        Blob mainContent = (Blob) document.getPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY);
+
+        // Check if the FILE_CONTENT_PROPERTY and the rendition are equals
+        assertEquals(expectedBlob.getString(), mainContent.getString());
+    }
+
+    @Test
+    @Deploy("org.nuxeo.coldstorage.test:OSGI-INF/test-coldstorage-rendition-contrib.xml")
+    public void shouldFailRenditionNotExist() throws IOException, OperationException {
+        DocumentModel document = createPictureDocumentModel(session);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        try {
+            moveContentToColdStorage(session, document);
+        } catch (NuxeoException e) {
+            assertEquals(SC_NOT_FOUND, e.getStatusCode());
+        }
+    }
+
+    @Test
+    public void shouldSelectDefaultVideoRendition() throws IOException, OperationException {
+        DocumentModel document = createVideoDocumentModel(session, "images/sample.mpg");
+
+        transactionalFeature.nextTransaction();
+
+        document.refresh();
+
+        // Retrieve the expected rendition
+        RenditionService renditionService = Framework.getService(RenditionService.class);
+        Rendition rendition = renditionService.getRendition(document, "MP4 480p");
+        Blob expectedBlob = rendition.getBlob();
+
+        moveContentToColdStorage(session, document);
+
+        transactionalFeature.nextTransaction();
+        document.refresh();
+
+        Blob mainContent = (Blob) document.getPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY);
+
+        // Check if the FILE_CONTENT_PROPERTY and the rendition are equals
+        assertEquals(expectedBlob.getString(), mainContent.getString());
+    }
+
+    private DocumentModel createPictureDocumentModel(CoreSession session) throws IOException {
+        DocumentModel document = session.createDocumentModel("/", "picture", "Picture");
+        Blob blob = Blobs.createBlob(FileUtils.getResourceFileFromContext("images/test.jpg"), "image/jpeg",
+                StandardCharsets.UTF_8.name(), "test.jpg");
+        document.setPropertyValue(ColdStorageConstants.FILE_CONTENT_PROPERTY, (Serializable) blob);
+        document = session.createDocument(document);
+        session.save();
+        return document;
+    }
+
+    private DocumentModel createVideoDocumentModel(CoreSession session, String path) throws IOException {
+        DocumentModel document = session.createDocumentModel("/", "video", VideoConstants.VIDEO_TYPE);
+        Blob blob;
+        try (InputStream inputStream = TestColdStorageRendition.class.getResourceAsStream("/" + path)) {
```

Comment:
Why use a stream here? Can't you use `Blobs.createBlob(FileUtils.getResourceFileFromContext(), ...)` as for the picture?

Reply:
I will try to use the same logic as before